enterprise/core-edge/src/test/java/org/neo4j/coreedge/scenarios/CoreToCoreCopySnapshotIT.java

```diff
@@ -251,4 +253,9 @@ private String string( int numberOfCharacters )
         }
         return s.toString();
     }
+
+    private CoreState getCoreState( CoreGraphDatabase follower )
+    {
+        return follower.getDependencyResolver().resolveDependency( CoreState.class );
```

Comment:
All that part of the code is badly designed and messed up, this is an attempt to make that code a bit better by reducing complexity.